app/controllers/post_favorites_controller.rb

```diff
@@ -0,0 +1,15 @@
+class PostFavoritesController < ApplicationController
+  before_action :member_only
+  respond_to :html
+
+  def index
+    @post = Post.find(params[:post_id])
+    query = User.includes(:user_status).joins("INNER JOIN favorites ON favorites.user_id = users.id")
```

Comment:
The scope on the user model for `favorites` isn't used by anything, so can be removed, leaving it as simple `has_many :favorites` and then this can be converted back into the simpler join form.